cpp/src/copying/contiguous_split.cu

```diff
@@ -1286,4 +2014,46 @@ std::vector<packed_table> contiguous_split(cudf::table_view const& input,
   return detail::contiguous_split(input, splits, cudf::get_default_stream(), mr);
 }
 
+chunked_pack::chunked_pack(cudf::table_view const& input,
+                           std::size_t user_buffer_size,
+                           rmm::mr::device_memory_resource* temp_mr)
+{
+  CUDF_EXPECTS(user_buffer_size >= desired_batch_size,
+               "The output buffer size must be at least 1MB in size");
+  // We pass `nullptr` for the first `mr` in `contiguous_split_state` to indicate
+  // that it does not allocate any user-bound data for the `chunked_pack` case.
+  state = std::make_unique<detail::contiguous_split_state>(
+    input, user_buffer_size, cudf::get_default_stream(), nullptr, temp_mr);
+}
+
+// required for the unique_ptr to work with a non-complete type (contiguous_split_state)
+chunked_pack::~chunked_pack() = default;
+
+std::size_t chunked_pack::get_total_contiguous_size() const
+{
+  return state->get_total_contiguous_size();
+}
+
+bool chunked_pack::has_next() const { return state->has_next(); }
+
+std::size_t chunked_pack::next(cudf::device_span<uint8_t> const& user_buffer)
+{
+  return state->contiguous_split_chunk(user_buffer);
+}
+
+std::unique_ptr<std::vector<uint8_t>> chunked_pack::build_metadata() const
+{
+  return state->build_packed_column_metadata();
+}
+
+std::unique_ptr<chunked_pack> chunked_pack::create(cudf::table_view const& input,
+                                                   std::size_t user_buffer_size,
+                                                   rmm::mr::device_memory_resource* temp_mr)
+{
+  // `temp_mr` could be a special memory resource to be used in situations when
+  // GPU memory is low and we want scratch and temporary allocations to happen from
+  // a small reserved pool of memory.
```

Comment:
I would put this information in the function's documentation in the header rather than inline here